lib/rules/at-rule-descriptor-value-no-invalid/index.cjs

```diff
@@ -0,0 +1,68 @@
+// NOTICE: This file is generated by Rollup. To modify it,
+// please instead edit the ESM counterpart and rebuild with Rollup (npm run build).
+'use strict';
+
+const cssTree = require('css-tree');
+const nodeFieldIndices = require('../../utils/nodeFieldIndices.cjs');
+const atKeywords = require('../../reference/atKeywords.cjs');
+const report = require('../../utils/report.cjs');
+const ruleMessages = require('../../utils/ruleMessages.cjs');
+const validateOptions = require('../../utils/validateOptions.cjs');
+
+const UNSUPPORTED_NESTING_REGEX = new RegExp(
+	`^((?!${[...atKeywords.nestingSupportedAtKeywords.values()].join('|')}).)*$`,
+	'i',
+);
+
+const ruleName = 'at-rule-descriptor-value-no-invalid';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (descriptor, value) =>
+		`Unexpected invalid value "${value}" for descriptor "${descriptor}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/at-rule-descriptor-value-no-invalid',
+};
+
+/** @type {import('stylelint').CoreRules[ruleName]} */
+const rule = (primary) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, { actual: primary });
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkAtRules(UNSUPPORTED_NESTING_REGEX, (atRule) => {
+			atRule.walkDecls((decl) => {
```

Comment:
```suggestion
		root.walkAtRules(UNSUPPORTED_NESTING_REGEX, (atRule) => {
		    if (!isStandardSyntaxAtRule(atRule)) return;
			atRule.walkDecls((decl) => {
			  if (!isStandardSyntaxDeclaration(decl)) return;
```

Let's return early for non-standard at-rules and decls.

See https://github.com/stylelint/stylelint/pull/8231

Reply:
Thanks for your review! I've applied it.
[Return early for non-standard at-rules and decls](https://github.com/stylelint/stylelint/pull/8211/commits/ded4ebbc264f11a6ec36efd9b6044b282c823dd6)